core/commands/p2p.go

```diff
@@ -129,14 +141,45 @@ Example:
 			return
 		}
 
-		if err := forwardLocal(n.Context(), n.P2P, n.Peerstore, proto, listen, target); err != nil {
+		if err := forwardLocal(n.Context(), n.P2P, n.Peerstore, proto, listen, targets); err != nil {
 			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
 		res.SetOutput(nil)
 	},
 }
 
+// parseIpfsAddr is a function that takes in addr string and return ipfsAddrs
+func parseIpfsAddr(addr string) ([]ipfsaddr.IPFSAddr, error) {
+	mutiladdr, err := ma.NewMultiaddr(addr)
+	if err != nil {
+		return nil, err
+	}
+	if strings.Contains(mutiladdr.String(), "/ipfs/Qm") {
```

Comment:
I wouldn't check for the `Qm` prefix as that won't work with Ed keys. Also, we want to migrate form `/ipfs/` to `/p2p/` in near future so for compatibility this should just be `if _, err := multiaddr.ValueForProtocol(ma.P_IPFS); err == nil {`